src/averageAges.js

```diff
@@ -55,7 +71,15 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const withKids = (onlyWithSon
+    ? people.filter(person => person.sex === 'm'
+    && people.find(mother => person.mother === mother.name))
+    : people.filter(person => people
+      .find(mom => person.mother === mom.name)));
```

Comment:
`people.find(mom => person.mother === mom.name)` - duplicated code can be in variable)